addons/sale_pdf_quote_builder/wizards/sale_pdf_quote_builder_dynamic_fields_wizard.py

```diff
@@ -0,0 +1,134 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import json
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import ValidationError
+
+from odoo.addons.sale_pdf_quote_builder.const import DEFAULT_FORM_FIELD_PATH_MAPPING
+
+
+class SalePDFQuoteBuilderDynamicFieldsWizard(models.TransientModel):
+    _name = 'sale.pdf.quote.builder.dynamic.fields.wizard'
+    _description = "Sale PDF Quote Builder Dynamic Fields Configurator Wizard"
+
+    current_form_fields = fields.Json()
+    dynamic_fields_wizard_line_ids = fields.One2many(
+        'sale.pdf.quote.builder.dynamic.fields.wizard.line', 'dynamic_fields_wizard_id'
+    )
+
+    @api.onchange('current_form_fields')
+    def _get_wizard_lines(self):
+        """ Add wizard lines containing existing mappings and form fields from the current document.
+        """
+        for wizard in self:
+            wizard.dynamic_fields_wizard_line_ids = [Command.clear()]
+            form_field_path_map = json.loads(self.env['ir.config_parameter'].sudo().get_param(
+                'sale_pdf_quote_builder.form_field_path_mapping', DEFAULT_FORM_FIELD_PATH_MAPPING
+            ))
+            current_form_fields = json.loads(
+                wizard.current_form_fields
+            ) if wizard.current_form_fields else {}
+
+            # Get the form fields not yet in the config parameters
+            lines_vals = []
+            for doc_type, form_fields in current_form_fields.items():
+                for form_field in form_fields:
+                    if form_field not in form_field_path_map[doc_type]:
+                        lines_vals.extend([
+                            {'document_type': doc_type, 'form_field': form_field, 'path': ''}
+                        ])
+            # Add existing values from the config parameter
+            lines_vals.extend([
+                {'document_type': doc_type, 'form_field': form_field, 'path': path}
+                for doc_type, mapping in form_field_path_map.items()
+                for form_field, path in mapping.items()
+            ])
```

Comment:
No, because we don't want the user to manually update the config parameter, and we don't want any menu action to do it. That's also why you can add lines that aren't in the document too, so that you could manage form field not in the document.